src/libsystemd-network/dhcp-static-lease.c

```diff
@@ -0,0 +1,94 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "alloc-util.h"
+#include "dhcp-internal.h"
+#include "dhcp-server-internal.h"
+#include "memory-util.h"
+#include "strv.h"
+#include "utf8.h"
+
+static sd_dhcp_static_lease *sd_dhcp_static_lease_free(sd_dhcp_static_lease *i) {
+        if (!i)
+                return NULL;
+
+        free(i->client_id.data);
+        return mfree(i);
+}
+
+static int dhcp_static_lease_new(
+        DHCPClientId *client_id, const struct in_addr *address, sd_dhcp_static_lease **ret) {
+
+        assert_return(ret, -EINVAL);
+        assert_return(address, -EINVAL);
+
+        sd_dhcp_static_lease *p = new (sd_dhcp_static_lease, 1);
+        if (!p)
+                return -ENOMEM;
+
+        *p = (sd_dhcp_static_lease){ .n_ref = 1, .client_id = *client_id, .address = le32toh(address->s_addr) };
```

Comment:
why cram this all in one line? just write:

```c
*p = (sd_dhcp_static_lease) {
        .n_ref = 1,
        .client_id…
}
```

Reply:
struct in_addr contains the IP address in network byte order, i.e. big endian. The `.address` field of `sd_dhcp_static_lease` is defined as be32_t, i.e. is declared to be big endian. Hence the le32toh conversion is wrong.

Reply:
Further explained [here](https://github.com/systemd/systemd/pull/15556#discussion_r484279791).